tools/debugging/extract_sp_exceptions.sh

```diff
@@ -42,94 +40,81 @@ else
-        [ -n "$run_number" ] || { echo 'Could not find run number!'; exit 1; }
-        download_nodes_logs "$scenario" "${current_server}" "${run_number}"
-        echo -e "\t - ${scenario}, run_number: ${run_number}"
-    done;
 
-    separator
+    (
+        cd "${SCENARIOS_DIR}"
+
+        ssh "$current_server" '
+        cd /var/lib/scenario-player;
+        source env.sh
+
+        cd /data/scenario-player/scenarios;
+        for scenario_path in "$SCENARIOS_DIR"/ci/"$SP"/*.yaml; do
+            scenario_file=$(basename "$scenario_path")
+            scenario=${scenario_file//.yaml}
+
+            run=$(cat "${scenario}"/run_number.txt)
+
+            find "$scenario"/node_"$run"_*
+            ls "$scenario"/scenario-player-run* -1 -t | head -1
+        done | tar zcf - -T -
+        ' | tar zxf -
+    )
 }
 
 function search_for_failures {
     mkdir -p "${DESTINATION_DIR}/errors/"
     echo -e "${BOLD}Looking for failures${RESET}"
-    scenarios_dir="${DESTINATION_DIR}/scenarios"
-
-    for scenario_dir in "${scenarios_dir}"/*; do
-        scenario=$(basename "${scenario_dir}")
-        print_bold "${scenario}"
-        separator
-
-        for node_dir in $(find "${scenario_dir}" -maxdepth 1 -type d -name "node_*"); do
-            result=$(gunzip -c ${node_dir}/*.log.gz | jq --tab 'select (.error!=null or .exception!=null)')
-            if [[ $result == "" ]]; then
-                result=$(cat "${node_dir}"/*.stderr | grep -v Starting | grep -v Stopped)
-            fi
-            if [[ $result != "" ]]; then
-                print_bold "- Found error in ${node_dir}"
-                echo -e "${result}"
-                echo "${result}" > "${DESTINATION_DIR}/errors/${scenario}.node.log.gz"
-            fi
-        done
-
-        separator
-        sp_error=$(gunzip -c "${scenario_dir}"/scenario-player-run_*.log.gz | jq 'select (.error!=null or .exception!=null)')
-
-        if [[ $sp_error != "" ]]; then
-            print_bold "- SP reported an error in ${scenario_dir}"
-
-            echo -e "${sp_error}"
-            echo "${sp_error}" > "${DESTINATION_DIR}/errors/${scenario}.sp.log"
-        fi
 
-        separator
+    for scenario_logs in $(find "${SCENARIOS_DIR}" -maxdepth 2 -type f -name "scenario-player*.gz"); do
+        scenario_successful=$(gunzip -c  "$scenario_logs" | jq 'select(.result == "success")')
+
+        if [ -z "${scenario_successful}" ]; then
+            scenario=$(basename "${scenario_logs}")
+            scenario_dir=$(dirname "${scenario_logs}")
+
+            separator
+            print_bold "Scenario ${scenario} failed"
+            gunzip -c  "$scenario_logs" | jq 'select(.result != null) | .message'
+            separator
+
+            for node_dir in $(find "${scenario_dir}" -maxdepth 1 -type d -name "node_*"); do
+                gunzip -c ${node_dir}/*.log.gz | jq --tab 'select (.error!=null or .exception!=null)'
+                cat "${node_dir}"/*.stderr | grep -v Starting | grep -v Stopped
```

Comment:
This changes the behavior of script, the error logs are not pushed to a folder anymore, this should be fine since running the script again is quite fast because the download is not redone.